### ISSUE TITLE
unsupported operand type(s) for *: 'int' and 'NoneType

### ISSUE THREAD

**Issue by dvnt-admin** [paul-gauthier/aider#475]
---aider.conf.yml file contents>

openai-api-key: sk-********************************
openai-api-base: https://api.openai.com/v1
dark-mode: True
pretty: True
stream: False

---terminal command>

PS C:\aider_projects\test>aider

---aider start output>
 
Aider v0.24.0
Model: gpt-4-0613 using diff edit format
Git repo: ..\..\..\..\.git with 0 files
Repo-map: using 1024 tokens

---aider command within aider>
>refactor

---any command results in the following>

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python312\Scripts\aider.exe\__main__.py", line 7, in <module>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python312\Lib\site-packages\aider\main.py", line 631, in main
    coder.run()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python312\Lib\site-packages\aider\coders\base_coder.py", line 391, in run
    new_user_message = self.run_loop()
                       ^^^^^^^^^^^^^^^
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python312\Lib\site-packages\aider\coders\base_coder.py", line 475, in run_loop
    return self.send_new_user_message(inp)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python312\Lib\site-packages\aider\coders\base_coder.py", line 530, in send_new_user_message
    interrupted = self.send(messages, functions=self.functions)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python312\Lib\site-packages\aider\coders\base_coder.py", line 658, in send
    self.show_send_output(completion)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python312\Lib\site-packages\aider\coders\base_coder.py", line 709, in show_send_output
    cost += completion_tokens * self.main_model.completion_price / 1000
            ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for *: 'int' and 'NoneTyp

================================================================

Any message sent to aider results in the above error now. I have been using aider for many months with no issues until today.

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

This bug was introduced by recent refactor of the way aider maintains info about all the OpenAI models and their costs, etc. The bug was only an issue when `--no-stream` is in effect, which is how it slipped by. 

I've fixed the issue and released `v0.24.1`. You should be able to pickup the fix by doing:

```
python -m pip install --upgrade aider-chat
```

**Comment by dvnt-admin** [comment]
this change from chatgpt held be over:

def show_send_output(self, completion):
        if self.verbose:
            print(completion)

        show_func_err = None
        show_content_err = None
        try:
            self.partial_response_function_call = completion.choices[0].message.function_call
        except AttributeError as func_err:
            show_func_err = func_err

        try:
            self.partial_response_content = completion.choices[0].message.content
        except AttributeError as content_err:
            show_content_err = content_err

        resp_hash = dict(
            function_call=self.partial_response_function_call,
            content=self.partial_response_content,
        )
        resp_hash = hashlib.sha1(json.dumps(resp_hash, sort_keys=True).encode())
        self.chat_completion_response_hashes.append(resp_hash.hexdigest())

        if show_func_err and show_content_err:
            self.io.tool_error(show_func_err)
            self.io.tool_error(show_content_err)
            raise Exception("No data found in openai response!")

        tokens = None
        if hasattr(completion, "usage") and completion.usage is not None:
            prompt_tokens = completion.usage.prompt_tokens
            completion_tokens = completion.usage.completion_tokens

            tokens = f"{prompt_tokens} prompt tokens, {completion_tokens} completion tokens"

            # Check and set a default value for completion_price if it is None
            if self.main_model.completion_price is None:
                self.main_model.completion_price = 0  # Assign a sensible default value here

            if self.main_model.prompt_price:
                cost = prompt_tokens * self.main_model.prompt_price / 1000
                cost += completion_tokens * self.main_model.completion_price / 1000
                tokens += f", ${cost:.6f} cost"
                self.total_cost += cost
                
        show_resp = self.render_incremental_response(True)
        if self.show_pretty():
            show_resp = Markdown(
                show_resp, style=self.assistant_output_color, code_theme=self.code_theme
            )
        else:
            show_resp = Text(show_resp or "<no response>")

        self.io.console.print(show_resp)

        if tokens is not None:
            self.io.tool_output(tokens)
~~~

thank you!
